src/sys/termios.rs

```diff
@@ -1,648 +1,782 @@
+///
+/// `cfgetispeed()` extracts the input baud rate from the given Termios structure.
 pub fn cfgetispeed(termios: &Termios) -> BaudRate {
-    unsafe {
-        ffi::cfgetispeed(termios).into()
-    }
+    unsafe { libc::cfgetispeed(termios).into() }
 }
 
+/// Get output baud rate (see
+/// [cfgetospeed(3p)](http://man7.org/linux/man-pages/man3/cfgetospeed.3p.html)).
+///
+/// `cfgetospeed()` extracts the output baud rate from the given Termios structure.
 pub fn cfgetospeed(termios: &Termios) -> BaudRate {
+    unsafe { libc::cfgetospeed(termios).into() }
+}
+
+/// Configures the port to something like the "raw" mode of the old Version 7 terminal driver (see
+/// [termios(3)](http://man7.org/linux/man-pages/man3/termios.3.html)).
+///
+/// `cfmakeraw()` configures the termios structure such that input is available character-by-
+/// character, echoing is disabled, and all special input and output processing is disabled. Note
+/// that this is a non-standard function, but is available on Linux and BSDs.
+pub fn cfmakeraw(termios: &mut Termios) {
     unsafe {
-        ffi::cfgetospeed(termios).into()
+        libc::cfmakeraw(termios);
     }
 }
 
+/// Set input baud rate (see
+/// [cfsetispeed(3p)](http://man7.org/linux/man-pages/man3/cfsetispeed.3p.html)).
+///
+/// `cfsetispeed()` sets the intput baud rate in the given Termios structure.
 pub fn cfsetispeed(termios: &mut Termios, baud: BaudRate) -> Result<()> {
-    Errno::result(unsafe {
-        ffi::cfsetispeed(termios, baud as speed_t)
-    }).map(drop)
+    Errno::result(unsafe { libc::cfsetispeed(termios, baud.into()) }).map(drop)
 }
 
+/// Set output baud rate (see
+/// [cfsetospeed(3p)](http://man7.org/linux/man-pages/man3/cfsetospeed.3p.html)).
+///
+/// `cfsetospeed()` sets the output baud rate in the given termios structure.
 pub fn cfsetospeed(termios: &mut Termios, baud: BaudRate) -> Result<()> {
-    Errno::result(unsafe {
-        ffi::cfsetospeed(termios, baud as speed_t)
-    }).map(drop)
+    Errno::result(unsafe { libc::cfsetospeed(termios, baud.into()) }).map(drop)
+}
+
+/// Set both the input and output baud rates (see
+/// [termios(3)](http://man7.org/linux/man-pages/man3/termios.3.html)).
+///
+/// `cfsetspeed()` sets the input and output baud rate in the given termios structure. Note that
+/// this is part of the 4.4BSD standard and not part of POSIX.
+pub fn cfsetspeed(termios: &mut Termios, baud: BaudRate) -> Result<()> {
+    Errno::result(unsafe { libc::cfsetspeed(termios, baud.into()) }).map(drop)
 }
 
+/// Return the configuration of a port
+/// [tcgetattr(3p)](http://man7.org/linux/man-pages/man3/tcgetattr.3p.html)).
+///
+/// `tcgetattr()` returns a Termios structure with the current configuration for a port. Modifying
+/// this structure *will not* reconfigure the port, instead the modifications should be done to
+/// the `Termios` structure and then the port should be reconfigured using `tcsetattr()`. Note
```

Comment:
Shouldn't this note about returning success if any parameters were successfully set belong in the comment block for `tcsetattr` instead of `tcgetattr`?